code/game/machinery/vending.dm

```diff
@@ -423,7 +503,7 @@
  */
 /obj/machinery/vending/proc/credit_purchase(var/target as text)
 	var/datum/transaction/T = new(currently_vending.price, target, "Purchase of [currently_vending.product_name]", name)
-	T.apply_to(vendor_account)
+	T.apply_to(vendor_department ? vendor_department : machine_vendor_account)
```

Comment:
I don't think you can apply a transaction directly to a department name like this, you'd need to fetch the relevant department account